### ISSUE TITLE
GitHub actions fail

### ISSUE THREAD

**Issue by ClaraBuettner** [openego/eGon-data#183]
All GitHub actions of PR that include the current dev fail with the following error: 

```
> 
>   ==================================== ERRORS ====================================
>   _ ERROR collecting .tox/py37-nocov-linux/lib/python3.7/site-packages/egon/data/airflow/dags/pipeline.py _
>   .tox/py37-nocov-linux/lib/python3.7/site-packages/egon/data/airflow/dags/pipeline.py:30: in <module>
>       airflow_db_connection()
>   .tox/py37-nocov-linux/lib/python3.7/site-packages/egon/data/db.py:108: in airflow_db_connection
>       cred = credentials()
>   .tox/py37-nocov-linux/lib/python3.7/site-packages/egon/data/db.py:28: in credentials
>       custom = config.paths(pid="*")[0]
>   E   IndexError: list index out of range
```

**Comment by gnn** [comment]
Ah, OK. I know what's happening here. The function `egon.data.db.airflow_db_connection` gets executed when `egon.data.airflow.dags.pipeline` gets imported, which is pretty much before anything else happens. This wasn't a problem when the connection parameters where read from packaged files. Now that they're read from the configuration file though, the problem is that this configuration file isn't yet generated at the point where `airflow_db_connection` tries to read the connection parameters, which triggers the error. I'll fix that.

**Comment by EvaWie** [comment]
There is another issue regarding 
custom = config.paths(pid="*")[0]
IndexError: list index out of range:
On my system the egon-data.pid-6751.configuration.yaml file disappears, when I stop airflow in terminal, but on @ClaraBuettner 's system it does not. That means for me, I cannot test my code in spyder, when airflow is not running.
Can we do something about it? Thanks for your help!

**Comment by gnn** [comment]
> On my system the egon-data.pid-6751.configuration.yaml file disappears, when I stop airflow in terminal,

That is intentional. The "egon-data.pid-N.configuration.yaml" file is specific to the process with process id N, i.e. a specifig `egon-data` invocation, and is intended to live only as long as this process.

> but on @ClaraBuettner 's system it does not.

Then this is a bug. Either there is still a running `egon-data` process to whom this file belongs, or a previous process failed to clean up and delete its file. Do you receive the message:
```
Ignoring supplied options. Found a configuration file belonging to a
different `egon-data` process. Using that one.
```
when running `egon-data`, @ClaraBuettner? Also, I tried to fix the issue at hand in PR #187 so could you please check if the branch `fixes/#183-make-build-pass-again` solves the issue you filed here?

> That means for me, I cannot test my code in spyder, when airflow is not running.
> Can we do something about it? Thanks for your help!

Sure. First, check whether your problem is still present on the `fixes/#183-make-build-pass-again` branch. If that's the case, post a detailed stack trace of the error, so that I can figure out where the line triggering the error is still used so that I can change it.

**Comment by gplssm** [comment]
I ran into `E   IndexError: list index out of range` as well, even when I executed a script directly from command-line (during development). Using `fixes/#183-make-build-pass-again` this is gone. Thanks :clap:

**Comment by ClaraBuettner** [comment]
> Then this is a bug. Either there is still a running `egon-data` process to whom this file belongs, or a previous process failed to clean up and delete its file. Do you receive the message:
> 
> ```
> Ignoring supplied options. Found a configuration file belonging to a
> different `egon-data` process. Using that one.
> ```
> 
> when running `egon-data`, @ClaraBuettner? Also, I tried to fix the issue at hand in PR #187 so could you please check if the branch `fixes/#183-make-build-pass-again` solves the issue you filed here?
> 

Yes, I received this message before. But when I check out to the branch of PR #187, it is gone and the egon-data.pid-....configuration.yaml file is deleted when I stop airflow and recreated when I start it again. So this is fixed.
